Rebase phase discards stderr and doesn't pre-check worktree cleanliness

## Problem

During shepherd run for #2585, the rebase phase (5.5) failed with:
```
[ERROR] rebase onto origin/main failed: Rebase failed (unknown conflicts)
```

But the PR was immediately mergeable on GitHub (`mergeStateStatus: CLEAN`). The shepherd exited with code 4 instead of completing the merge.

### Root Cause

Two issues in `attempt_rebase()` (`loom-tools/src/loom_tools/common/git.py:353-396`):

**1. `result.stderr` is discarded.** The function captures stderr from `git rebase` but never reads it:
```python
result = run_git(["rebase", base], cwd=cwd, check=False)
if result.returncode == 0:
    return True, ""
# ... tries to find conflicting files via --diff-filter=U
# ... but never reads result.stderr
detail = "Rebase failed (unknown conflicts)"  # Generic fallback
```

The actual git error message (e.g., "Cannot rebase: You have unstaged changes") is lost.

**2. No dirty-worktree pre-check.** `git rebase` refuses to start if there are uncommitted changes. After the doctor phase, the worktree may have dirty state. The rebase phase doesn't verify cleanliness before attempting the rebase. When `git rebase` refuses to start, there are no unmerged files for `--diff-filter=U` to find, so the error falls through to the generic "unknown conflicts" message.

### Affected Code

- `loom-tools/src/loom_tools/common/git.py` — `attempt_rebase()` (line 353)
- `loom-tools/src/loom_tools/shepherd/phases/rebase.py` — `RebasePhase.run()` (line 34)
- `loom-tools/src/loom_tools/shepherd/phases/doctor.py` — doesn't clean worktree on exit

### Suggested Fixes

1. **Include `result.stderr` in the failure detail:**
   ```python
   stderr_info = result.stderr.strip() if result.stderr else ""
   if conflicting:
       detail = f"Conflicting files:\n{conflicting}"
   elif stderr_info:
       detail = f"Rebase failed: {stderr_info}"
   else:
       detail = "Rebase failed (unknown conflicts)"
   ```

2. **Add dirty-worktree pre-check** before attempting rebase — check `has_uncommitted_changes()` and either stash or report clearly.

3. **Consider `git rebase --autostash`** which automatically stashes uncommitted changes before rebasing and re-applies them after.

### Impact

The shepherd exits with error code 4 (rebase failure) even when the PR is cleanly mergeable, requiring manual intervention to complete the merge. The misleading "unknown conflicts" message makes debugging harder.

--- a/loom-tools/src/loom_tools/common/git.py
+++ b/loom-tools/src/loom_tools/common/git.py
@@ -400,7 +400,7 @@ def attempt_rebase(
         conflicting files.
     """
     try:
-        result = run_git(["rebase", base], cwd=cwd, check=False)
+        result = run_git(["rebase", "--autostash", base], cwd=cwd, check=False)
         if result.returncode == 0:
             return True, ""
 
@@ -415,7 +415,13 @@ def attempt_rebase(
         # Abort the failed rebase
         run_git(["rebase", "--abort"], cwd=cwd, check=False)
 
-        detail = f"Conflicting files:\n{conflicting}" if conflicting else "Rebase failed (unknown conflicts)"
+        stderr_info = result.stderr.strip() if result.stderr else ""
+        if conflicting:
+            detail = f"Conflicting files:\n{conflicting}"
+        elif stderr_info:
+            detail = f"Rebase failed: {stderr_info}"
+        else:
+            detail = "Rebase failed (unknown conflicts)"
         return False, detail
     except Exception as exc:
         # Safety: abort any in-progress rebase

